encoders/src/test/scala/sql/EncoderDerivationSpec.scala

```diff
@@ -398,3 +402,33 @@ class EncoderDerivationSpec extends munit.FunSuite with SparkSqlTesting:
       )
     )
   }
+
+  test("Custom encoder to a primitive type wrapped in an Option") {
+    // Defining some custom Serializer and Deserializer for Instant to LongType
+    given Serializer[Instant] with
```

Comment:
the Instant serializer is already available in `Serializer.scala` and `Deserializer.scala` - so why do you think it is necessary here? Or is this the correct implementation.

Reply:
This is a custom `Instant` serializer that maps to a `LongType`. It's a real world example from my work where we needed to keep back compat (the Long type) with the table but would prefer using an `Instant` in the code.
In this scenario, the code in main does not work.
It would have been the same with `case class Ex(value: Long)` and a custom serializer that maps to a `LongType`.

Reply:
The use case of adding Encoders in our code, which was not possible with composition of regular spark encoders, is interesting I think.
I wonder if it would be useful to offer in the api utility methods.
In this case we know that an encoder for `Long` exists, so it would be easier if we had a utility like 
```scala
given [A, B: Encoder](f: A => B): Encoder[A] = ???
```
similar to `contramap`in other library